### DISCUSSION TITLE
Clarify "recommended" vs. "best-practices" presets on web page

### DISCUSSION THREAD

**Discussion by lathspell** [renovatebot/renovate#23657] [Suggest an Idea]
### Tell us more.

The preset configs on https://docs.renovatebot.com/presets-config/ list both, a "best-practices" and a "recommended".
As newbie it's unclear to me which one I should use as base. 

I presume that recommended is for a broader audience and works for more projects and best-practices is more opinionated on how the Renovate devs suggest teams should handle their dependency upgrades. This though is not explained and users do not understand how exactly they differ. Maybe it would also help if "best-practices" would itself be based on "recommended" and not "base"?

**Comment by HonkingGoose** [comment]
> 'Why indeed should we welcome you, Master Stormcrow? _Láthspell_ I name you, I'll-news; and ill news is an ill guest they say.'
>
> _Grima Wormtongue, Lord of the Rings, The Two Towers_

@lathspell I'm happy with your tidings, I'm not under Grima Wormtongue's spell. 😉 

## Which preset should I use?

> The preset configs on https://docs.renovatebot.com/presets-config/ list both, a "best-practices" and a "recommended".
> As newbie it's unclear to me which one I should use as base.

- `config:recommended` is for newcomers, that want a basic Renovate config, that's "good enough".
- `config:best-practices` is for those users who want to follow our recommendations for upgrades.

I recommend that you start with `config:recommended`. You can always use `config:best-practices` later, once you have more experience with Renovate, or want to follow our best practices. 😉 

### Explain better?

> I presume that recommended is for a broader audience and works for more projects and best-practices is more opinionated on how the Renovate devs suggest teams should handle their dependency upgrades.

I'm working on a PR that explains our thinking behind making the best practice preset:

- #22233

> This though is not explained and users do not understand how exactly they differ.

There's a _short_ description next to each preset, that gives a one-line summary:

> `config:recommended`
> Recommended configuration for most users. It does not matter what programming language you use.

And:

> `config:best-practices`
> Preset with best practices from the Renovate maintainers.

I agree the description is short. But that's because I don't want to overload the `description` field.

To find out precisely what a preset does, you'll need to read the docs for each of the config options in the preset.

## `config:best-practices` should extend from `config:recommended`

> Maybe it would also help if "best-practices" would itself be based on "recommended" and not "base"?

Good catch, I saw this too, see this issue:

- #23326

It's not a problem, because Renovate converts `config:base` to `config:recommended` behind the scenes. But we should clean it up now that we've released `v36` of Renovate. 😄

**Reply by HonkingGoose** [reply]
I'm also updating the `description` of the `config:best-practice` preset:

- #23773
